### PR TITLE
Docs: Add a section in Servers for setting extra capabilities

### PR DESCRIPTION
**Pull request by KengoA**
**What this PR does / why we need it**:

Add a section in `Servers` page explaining how to set `extraCapabilities` for servers and `requirements` for models.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/source/contents/servers/index.md [hunk 1]
```diff
@@ -22,6 +22,20 @@ There is a default capabilities for each server as follows:
   :end-before: SELDON_OVERCOMMIT_PERCENTAGE
   ```
 
+## Extra Capabilities
+Servers can be annotated with the `extraCapabilities` field to indicate custom configurations (e.g. Python dependencies). For instance:
```
Comment (agrski) [inline docs/source/contents/servers/index.md:26]: Given this is a k8s resource, probably best to avoid `annotated` as that implies `metadata.annotations`

Comment (agrski) [inline docs/source/contents/servers/index.md:26]: :thought_balloon: It'd be good to mention that the `ServerConfig` defines the default `capabilities` and this field allows a particular `Server` does not override that, but rather extends it with more capabilities

Comment (KengoA) [reply docs/source/contents/servers/index.md]: Agreed. Fixed wording in https://github.com/SeldonIO/seldon-core/pull/4718/commits/1e7bb30d7492d9b4fbd56f8de232f13cb2400b9f.

Comment (KengoA) [reply docs/source/contents/servers/index.md]: Good point. Added explanations on capabilities in https://github.com/SeldonIO/seldon-core/pull/4718/commits/1e7bb30d7492d9b4fbd56f8de232f13cb2400b9f.

File: samples/models/extra-capabilities.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: mlops.seldon.io/v1alpha1
+kind: Model
+metadata:
+  name: custom-model
```
Comment (agrski) [inline samples/models/extra-capabilities.yaml:4]: :thought_balloon: Really minor, but this isn't a custom model in any sense, just one that adds extra caps

**Review by agrski — APPROVED** [review]
We should probably revisit this page and add more detail here, as it's a bit brief overall.

This PR is a good step towards that :rocket:

File: samples/models/extra-capabilities.yaml (hunk 2 above)
Comment (KengoA) [reply samples/models/extra-capabilities.yaml]: Fixed this along with the filename in https://github.com/SeldonIO/seldon-core/pull/4718/commits/a7b4f0513dcc4e2e425ef5e1e5984438c3da43d9.

File: docs/source/contents/servers/index.md [hunk 3]
```diff
@@ -22,6 +22,20 @@ There is a default capabilities for each server as follows:
   :end-before: SELDON_OVERCOMMIT_PERCENTAGE
   ```
 
+## Extra Capabilities
+Servers can be set up with the `extraCapabilities` field to indicate custom configurations (e.g. Python dependencies). For instance:
+
+```{literalinclude} ../../../../samples/servers/mlserver-extra-capabilities.yaml
+:language: yaml
+```
+Note that `serverConfig: mlserver` will provide default capabilities for MLServer as shown above, and the values specified in `extraCapabilities` are appended to them to create a single list of capabilities.
```
Comment (agrski) [inline docs/source/contents/servers/index.md:31]:
```suggestion
Note that `serverConfig: mlserver` will provide default capabilities for MLServer as shown above, and the values specified in `extraCapabilities` are appended to them to create a single list of capabilities for this `Server`.
```

Comment (agrski) [reply docs/source/contents/servers/index.md:31]:
```suggestion
This server, `mlserver-extra`, inherits a default set of capabilities via `serverConfig: mlserver`.
These defaults are discussed above.
The `extraCapabilities` are appended to these to create a single list of capabilities for this server.
```
Alternative suggestion that perhaps read better?

File: docs/source/contents/servers/index.md [hunk 4]
```diff
@@ -22,6 +22,20 @@ There is a default capabilities for each server as follows:
   :end-before: SELDON_OVERCOMMIT_PERCENTAGE
   ```
 
+## Extra Capabilities
+Servers can be set up with the `extraCapabilities` field to indicate custom configurations (e.g. Python dependencies). For instance:
+
+```{literalinclude} ../../../../samples/servers/mlserver-extra-capabilities.yaml
+:language: yaml
+```
+Note that `serverConfig: mlserver` will provide default capabilities for MLServer as shown above, and the values specified in `extraCapabilities` are appended to them to create a single list of capabilities for this `Server`.
```
Comment (KengoA) [inline docs/source/contents/servers/index.md:31]: As suggested by @agrski 
```suggestion
This server, `mlserver-extra`, inherits a default set of capabilities via `serverConfig: mlserver`.
These defaults are discussed above.
The `extraCapabilities` are appended to these to create a single list of capabilities for this server.
```

File: docs/source/contents/servers/index.md (hunk 3 above)
Comment (KengoA) [reply docs/source/contents/servers/index.md]: Yes the second suggestion reads better. I added a new commit suggestion below myself as I couldn't apply it to the outdated file.

**Review by cliveseldon — APPROVED** [review]

**Pull request merged**